geopandas/array.py

```diff
@@ -84,6 +84,31 @@ def _check_crs(left, right, allow_none=False):
     return True
 
 
+def _crs_mismatch_warn(left, right, stacklevel=1):
+    if left.crs:
+        left_srs = left.crs.to_string()
+        left_srs = left_srs if len(left_srs) <= 50 else " ".join([left_srs[:50], "..."])
+    else:
+        left_srs = None
+    if right.crs:
+        right_srs = right.crs.to_string()
+        right_srs = (
+            right_srs if len(right_srs) <= 50 else " ".join([right_srs[:50], "..."])
+        )
+    else:
+        right_srs = None
+    warnings.warn(
+        "CRS mismatch between the CRS of left geometries "
+        "and the CRS of right geometries.\n"
+        "Use `GeoSeries.to_crs()` to reproject one of "
```

Comment:
```suggestion
        "Use `to_crs()` to reproject one of "
```

(to keep it generic, as one can also work with GeoDataFrame?)